lib/floe/workflow/state.rb

```diff
@@ -19,27 +19,28 @@ def build!(workflow, name, payload)
         end
       end
 
-      attr_reader :workflow, :comment, :name, :type, :payload
+      attr_reader :workflow, :comment, :name, :next_state, :type, :payload
 
       def initialize(workflow, name, payload)
-        @workflow = workflow
-        @name     = name
-        @payload  = payload
-        @end      = !!payload["End"]
-        @type     = payload["Type"]
-        @comment  = payload["Comment"]
+        @workflow   = workflow
+        @name       = name
+        @payload    = payload
+        end_state   = !!payload["End"]
+        @type       = payload["Type"]
+        @comment    = payload["Comment"]
+        @next_state = payload["Next"] unless end_state
       end
 
       def end?
-        @end
```

Comment:
I think you're trying to remove the "end" ivar, but the reason we had it was because it's an actual value in the payload - so I'm not sure about removing it.  Personally I was trying to stick to that "if it's in the payload, we save the raw value in the ivar" to keep those concept separate, but perhaps that's too pedantic.

@agrare Thoughts?

This kind of started changing with https://github.com/ManageIQ/floe/pull/62 - that original code was intentional, because "End" doesn't exist in the payload, so it didn't seem right to hardcode an `@end` which to me implies it is in the payload.  If we want to move away from that pattern, I'm good with that, but want to make sure that it's the direction we want to go.

Reply:
Yeah I think this is a similar issue to the input_path/output_path issue, while it would be convenient to handle this in the base `State` class not every type of state has Next defined.  So having `@next` in the base `State` class breaks the pattern we've had of keeping very close to the ASL spec for our class definitions.

Reply:
so I considered End and Next to be redundant.
But I appreciate making the two more similar

At a high level, all states have a `next`/`end?` concept, with `choice` and `task` being the only states that have a variable next.

If we want to go back to `@next` and `@end` and have each node parsing those concepts that works, too.

Reply:
as for input_path / output_path. having an intermediate node that handles path seemed like a good alternative.